### PR TITLE
Summernote Refile update

### PR DESCRIPTION
**Pull request by garyanewsome**
###### Use refile to upload summernote blog images - Issue #41 
- adapted Summernote to use Refile instead of its own base64 default

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -31,9 +32,21 @@ $(function(){
...
+       $summernote.summernote('insertNode', imgNode);
+     }
+   }
+ });
+});
+
+// summernote.image.upload
+$('#summernote').on('summernote.image.upload', function(we, files) {
+ // upload image to server and create imgNode...
+ $summernote.summernote('insertNode', imgNode);
```
Comment (ninjapanzer) [inline app/assets/javascripts/application.js]: short a space on indentation

File: app/models/post.rb [hunk 2]
```diff
@@ -1,2 +1,4 @@
 class Post < ActiveRecord::Base
+  has_many :post_attachment
```
Comment (ninjapanzer) [inline app/models/post.rb]: should be plural

File: app/models/post_attachment.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class PostAttachment < ActiveRecord::Base
+  belongs_to :posts
```
Comment (ninjapanzer) [inline app/models/post_attachment.rb]: should be singular

File: db/migrate/20160930155055_add_post_image_to_post_attachments.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class AddPostImageToPostAttachments < ActiveRecord::Migration
+  def change
+    add_column :post_attachments, :post_image, :string
```
Comment (ninjapanzer) [inline db/migrate/20160930155055_add_post_image_to_post_attachments.rb]: probably should be `post_image_id`

Comment (ninjapanzer) [inline db/migrate/20160930155055_add_post_image_to_post_attachments.rb]: Also add an index for the post_image_id

File: app/controllers/posts_controller.rb [hunk 5]
```diff
@@ -19,17 +19,23 @@ def show
...
-    if @post.save
-      flash[:notice] = "Successfully created post!"
-      redirect_to post_path(@post)
-    else
-      flash[:alert] = "Error creating new post!"
-      render :new
+    respond_to do |i|
+      if @post.save
+        params[:post_attachments].each do |i|
+          @post_attachment = @post.post_attachments
```
Comment (ninjapanzer) [inline app/controllers/posts_controller.rb]: I don't think you have to do this at all

Comment (ninjapanzer) [inline app/controllers/posts_controller.rb]: Or a save should be involved here so maybe what you really want is to do
```
@post.post_attachments = i
@post.save
```

File: app/assets/javascripts/posts.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
+$(function() {
+  var deleteFile, ready, sendFile;
```
Comment (ninjapanzer) [inline app/assets/javascripts/posts.js]: just put a var before each one. This is a style thing but hoisting variable is passé.

File: app/assets/javascripts/posts.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  sendFile = function(that, file) {
+    var data;
+    data = new FormData;
+    data.append('image[image]', file);
+    return $.ajax({
+      data: data,
+      type: 'POST',
+      url: '/images',
```
Comment (ninjapanzer) [inline app/assets/javascripts/posts.js]: This is the key part to saving. The url has to match to a controller for `post_images` so along with your model you need a `post_images_controller`

**Comment by garyanewsome** [comment]
screen shot 2016-09-30 at 5 04 27 pm

**Comment by ninjapanzer** [comment]
Updated with working create

out

**Comment by ninjapanzer** [comment]
out

**Pull request merged**